src/smt/expr_names.cpp

```diff
@@ -0,0 +1,40 @@
+/*********************                                                        */
+/*! \file expr_names.cpp
+ ** \verbatim
+ ** Top contributors (to current version):
+ **   Andrew Reynolds
+ ** This file is part of the CVC4 project.
+ ** Copyright (c) 2009-2020 by the authors listed in the file AUTHORS
+ ** in the top-level source directory) and their institutional affiliations.
+ ** All rights reserved.  See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** \brief Utility for maintaining expression names.
+ **/
+
+#include "smt/expr_names.h"
+
+namespace CVC4 {
+namespace smt {
+
+ExprNames::ExprNames(context::UserContext* u)
+    : d_exprNames(u)
+{
+}
+ExprNames::~ExprNames() {}
+
+void ExprNames::setExpressionName(Node e, const std::string& name) {
+  d_exprNames[e] = name;
+}
+
+bool ExprNames::getExpressionName(Node e, std::string& name) const {
+  context::CDHashMap< Node, std::string, NodeHashFunction >::const_iterator it = d_exprNames.find(e);
```

Comment:
Knowing full well that you might not like it :-)
```suggestion
  auto it = d_exprNames.find(e);
```

Reply:
I don't like it, but I don't really care about this code.  I'll pick my battles ;)